### PR TITLE
Add Developer Documentation

### PR DESCRIPTION
**Pull request by itamargiv**
This PR adds the initial developer documentation for this application.

Bug: [T285331](https://phabricator.wikimedia.org/T285331)

### CODE REVIEW

**Review by Silvan-WMDE — APPROVED** [review]
Haven't followed this step-by-step, but I trust that it works and it looks good to me.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+```
+**Possible Explanation:**
+
+After following the steps in the [Working with OAuth section](#working-with-oauth) you might get this error. It means your OAuth Consumer Registration is still being processed and you have to wait for the consumer key and consumer secret to be approved.
+
+This error might be caused as well if you have entered the wrong consumer credentials, or a wrong APP_URL. 
+
+**Solution:**  Wait a few hours and try again.
+
+## Sources
```
Comment (Silvan-WMDE) [inline docs/README.md:209]: anchor missing

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+Error in retrieving temporary credentials.
+http://localhost/auth/login
+```
+**Possible Explanation:**
+
+After following the steps in the [Working with OAuth section](#working-with-oauth) you might get this error. It means your OAuth Consumer Registration is still being processed and you have to wait for the consumer key and consumer secret to be approved.
+
+This error might be caused as well if you have entered the wrong consumer credentials, or a wrong APP_URL. 
+
+**Solution:**  Wait a few hours and try again.
```
Comment (itamargiv) [inline docs/README.md:207]:
```suggestion
**Solution:**
1. Make sure your app key and secert are correctly set to your consumer key and secert in the `.env file`
1. Make sure that your `APP_URL` is set correctly to your localhost address and the port your application is running on (if running on a non default port)
1. Double check to see that the WMF has sent an email to approve your localhost oauth consumer, if they have not responses yet, please wait patiently, as the review process might take a few hours.
```

**Review by itamargiv — COMMENTED** [review]
@guergana One tiny suggestion :) otherwise looks great! Thanks for the addition!

File: docs/README.md (hunk 1 above)
Comment (itamargiv) [reply docs/README.md:209]: :+1:

File: docs/README.md (hunk 2 above)
Comment (guergana) [reply docs/README.md:207]: spelling mistake: secret instead of secert

Comment (itamargiv) [reply docs/README.md:207]: I think I might've fixed it

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+1. Register your local testing app against wikidata in the following link:
+
+    [OAuth Consumer Registration](https://meta.wikimedia.org/wiki/Special:OAuthConsumerRegistration/propose)
+
+1. Fill in the following fields in the form as follows (see [screenshot](gallery/Screenshot_2021-06-28_at_18-06-22_OAuth_consumer_registration_-_Meta.md) for example):
+    - **Application name:** Make sure to include ***test*** in the name of the application, so that it is apparent for reviewers in the Wikimedia Foundation (WMF) that your consumer is meant for testing purposes.
+    - **OAuth protocol version:** Leave this field as is. This application uses OAuth1.0a.
+    - **Application description:** Fill in a description for your application, to make review easier.
+    - **OAuth "callback" URL:** Make sure to fill this in with a ***localhost*** address, with the port you would be running this application on (if not using the default port). This should be identical to the `APP_URL` in your `.env` file.
```
Comment (guergana) [inline docs/README.md:110]:
```suggestion
    - **OAuth "callback" URL:** Make sure to fill this in with a ***localhost*** address, with the port you would be running this application on (if not using the default port). This should be identical to the `APP_URL` in your `.env` file.
    Please also make sure you tick the checkbox next to "Allow consumer to specify a callback in requests and use "callback" URL above as required prefix."
```

**Review by Silvan-WMDE — APPROVED** [review]

**Pull request merged**